src/scripts/ci_build.py

```diff
@@ -271,20 +275,21 @@ def determine_flags(target, target_os, target_cpu, target_cc, cc_bin,
         if target_os in ['osx', 'ios']:
             flags += ['--with-commoncrypto']
 
-        if target == 'coverage':
+        if target in ['coverage', 'shared', 'static', 'amalgamation']:
             flags += ['--with-boost']
+            if target_cc == 'clang':
+                # make sure clang ignores warnings in boost headers
+                flags += ["--extra-cxxflags=--system-header-prefix=boost/"]
 
-        if target_os == 'windows' and target in ['shared', 'static']:
-            # ./configure.py needs extra hand-holding for boost on windows
-            boost_root = os.environ.get('BOOST_ROOT')
-            boost_libs = os.environ.get('BOOST_LIBRARYDIR')
-            boost_system = os.environ.get('BOOST_SYSTEM_LIBRARY')
+            if target_os in ['windows', 'mingw']:
+                # ./configure.py needs boost's location on Windows
+                assert 'BOOST_INCLUDEDIR' in os.environ, "Windows needs to know where to find boost (via BOOST_INCLUDEDIR)"
+                flags += ['--with-external-includedir', os.environ.get('BOOST_INCLUDEDIR')]
 
-            if boost_root and boost_libs and boost_system:
-                flags += ['--with-boost',
-                          '--with-external-includedir', boost_root,
-                          '--with-external-libdir', boost_libs,
-                          '--boost-library-name', boost_system]
+            if target_os == 'mingw':
+                # apparently mingw needs this legacy socket library version for reasons
+                # as per: https://stackoverflow.com/questions/38770895/how-to-fix-undefined-reference-to-getacceptexsockaddrs-boost-asio-in-clion#comment105791579_38771260
+                flags += ['--ldflags=-static -lwsock32']
```

Comment:
Technically, we could also add this to the `boost/info.txt` as a global dependency. But since `wsock32` is a somewhat legacy library I was hoping that we need to add this here more as a workaround than an actual required library. I.e. the windows-build of MinGW doesn't need it to pass.